applications/GeoMechanicsApplication/custom_conditions/general_U_Pw_diff_order_condition.cpp

```diff
@@ -272,21 +259,9 @@ void GeneralUPwDiffOrderCondition::CalculateAndAddConditionForce(VectorType& rRi
 
 Condition::DofsVectorType GeneralUPwDiffOrderCondition::GetDofs() const
 {
-    Condition::DofsVectorType result;
-
-    for (const auto& r_node : GetGeometry()) {
-        result.push_back(r_node.pGetDof(DISPLACEMENT_X));
-        result.push_back(r_node.pGetDof(DISPLACEMENT_Y));
-        if (GetGeometry().WorkingSpaceDimension() == 3) {
-            result.push_back(r_node.pGetDof(DISPLACEMENT_Z));
-        }
-    }
-
-    const auto water_pressure_dofs =
-        Geo::DofUtilities::ExtractDofsFromNodes(*mpPressureGeometry, WATER_PRESSURE);
-    result.insert(result.end(), water_pressure_dofs.begin(), water_pressure_dofs.end());
-
-    return result;
+    const auto r_geometry = this->GetGeometry();
```

Comment:
We're missing an `&` here:
```suggestion
    const auto& r_geometry = this->GetGeometry();
```

Reply:
added &